core/network/endpoint_bindings.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package network
+
+import "github.com/juju/errors"
+
+// MapWithSpaceNames returns the current bindingMap with space names rather than ids.
+func MapBindingsWithSpaceNames(bindingsMap map[string]string, lookup SpaceInfos) (map[string]string, error) {
+	// Handle the fact that space name lookup can be nil or empty.
+	if lookup == nil || (len(bindingsMap) > 0 && len(lookup) == 0) {
+		return nil, errors.NotValidf("empty space lookup")
+	}
+
+	retVal := make(map[string]string, len(bindingsMap))
+
+	// Assume that bindings is always in space id format due to
+	// Bindings constructor.
+	for k, v := range bindingsMap {
+		spaceInfo := lookup.GetByID(v)
+		if spaceInfo == nil {
+			return nil, errors.NotFoundf("space with ID %q", v)
```

Comment:
`return nil, errors.Errorf("space with ID %q does not exist", v).Add(coreerrors.NotFound)